Game/asteroid_view_model.gd

```diff
@@ -14,21 +16,35 @@ var building_on_cursor: BuildingResource:
 
 var in_build_mode: bool:
 	get:
-		return building_on_cursor != null
+		return building_on_cursor != Types.Building.NONE
 
 var mouse_state := MouseState.HOVERING
 # default value is null
 var mouse_tile_map_pos: TileMapPosition
 
 ## Whether the ores_layout in Model was updated this frame.
 var ores_layout_dirty: bool = false
+## Whether the buildings list in Model was updated this frame.
+var buildings_dirty: bool = false
+
 
 func _ready():
 	Model.ores_layout_updated.connect(_on_ores_layout_updated)
+	Model.buildings_updated.connect(_on_buildings_updated)
+
 
 func _process(_delta: float) -> void:
```

Comment:
Add a comment in this specifying that htis is visual information and thus uses _process instead of _physics_process (In case we get confused later on)